engine/src/main/java/org/hibernate/validator/internal/metadata/core/ConstraintHelper.java

```diff
@@ -139,6 +139,7 @@
 	private final Map<Class<? extends Annotation>, List<? extends Class<?>>> builtinConstraints;
 
 	private final ValidatorClassMap validatorClasses = new ValidatorClassMap();
+	private ConcurrentMap<Class<? extends Annotation>, Object> overridenAnnotations = newConcurrentHashMap();
```

Comment:
The ConstraintHelper is used both for populating the default validators for each annotation and for later override, either through XML configuration or through the programmatic API.
Since the `ConstraintHelper` is the most high-level place where both the XML parser and the programmatic API add "constraint definitions", it seems the best place to add the checks that will throw an exception whenever a user "overrides" the default validators more than once (assuming it's what we want).
And to do that, we must store somewhere the information "this annotation has been overridden", because just checking the existence of an entry in the map is not enough: it may already contain default values. Also, I'm using a map here because it's simpler than finding some thread-safe implementation of Set.
I could have added some complex checks, comparing the content of `validatorClasses` and of `builtinConstraints`, but this does not seem more elegant to me, and it would not handle the case where a user overrides the default with... exactly the default validators.

I will add a comment and fix the typo ("overrid**d**en", missing "d")